controllers/auth.js

```diff
@@ -0,0 +1,12 @@
+const User = require('../models/user')
+const { StatusCodes } = require('http-status-codes')
+
+const register = async (req, res) => {
+    const user = await User.create({ ...req.body })
+    const token = user.createJWT()
+    res.status(StatusCodes.CREATED).json({ user: { name: user.name }, token })
```

Comment:
Re check the stuff i sent you about the folder structure, logic here should be inside services. 

Also i dont see this actually saving to the db anywhere?
theere should be a user.save() on the schema, you can check us9ing the mongo cli if its in the db.